### PR TITLE
Make the python bufr2ioda driver more generic

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
Rather than each new ob type being manually added to the list, we can use `glob.glob` to get the list of available observation types.

### CODE REVIEW

**Review by nicholasesposito — APPROVED** [review]
Makes sense

File: ush/ioda/bufr2ioda/run_bufr2ioda.py [hunk 1]
```diff
@@ -30,8 +31,9 @@ def bufr2ioda(current_cycle, RUN, DMPDIR, config_template_dir, COM_OBS):
     }
 
     # Specify observation types to be processed by a script
-    BUFR_py = ["satwind_amv_goes", "satwind_scat", "adpupa_prepbufr", "adpsfc_prepbufr", "sfcshp_prepbufr", "acft_profiles_prepbufr",
-               "gpsro_bufr", "conventional_prepbufr_ps"]
+    BUFR_py_file = glob.glob(os.path.join(USH_IODA, 'bufr2ioda_*.py'))
+    BUFR_py_files = [os.path.basename(f) for f in BUFR_py_files]
```
Comment (RussTreadon-NOAA) [inline ush/ioda/bufr2ioda/run_bufr2ioda.py:35]: Should the second occurrence of `BUFR_py_files` be `BUFR_py_file`?  That is, should this line read

`BUFR_py_files = [os.path.basename(f) for f in BUFR_py_file]`

Comment (CoryMartin-NOAA) [reply ush/ioda/bufr2ioda/run_bufr2ioda.py:35]: good catch, I would advocate for the opposite, make the first line `BUFR_py_files`. I'll fix.

**Review by RussTreadon-NOAA — APPROVED** [review]
Tested updated `ush/ioda/bufr2ioda/run_bufr2ioda.py` in 2021080100 gdasprepatmiodaobs.   Job ran to completion.  

`ush/ioda/bufr2ioda` contained
```
bufr2ioda_adpsfc_prepbufr.py
bufr2ioda_adpupa_prepbufr.py
bufr2ioda_satwind_amv_goes.py
bufr2ioda_satwind_scat.py
bufr2ioda_sfcshp_prepbufr.py
```
`comrot/gdas_eval_satwind_JEDI/gdas.20210801/00/obs` contained
```
gdas.t00z.ADPSFC.prepbufr.nc
gdas.t00z.ascatw.ascat_metop-a.tm00.nc
gdas.t00z.ascatw.ascat_metop-b.tm00.nc
gdas.t00z.prepbufr.nc
gdas.t00z.satwnd.abi_goes-16.tm00.nc
gdas.t00z.satwnd.abi_goes-17.tm00.nc
gdas.t00z.SFCSHP.prepbufr.nc
```

Approve.

**Pull request merged**